src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -849,86 +817,49 @@ private void createAndSetStatistics()
     {
         final int citizensSize = townHall.getColony().getCitizens().size();
 
-        int workers = 0;
-        int builders = 0;
-        int deliverymen = 0;
-        int miners = 0;
-        int fishermen = 0;
-        int guards = 0;
-        int lumberjacks = 0;
-        int farmers = 0;
-        int bakers = 0;
-        int cooks = 0;
-
+        final Map<String, Integer> jobCountMap = new HashMap<>();
         for (@NotNull final CitizenDataView citizen : citizens)
         {
-            switch (citizen.getJob())
-            {
-                case COM_MINECOLONIES_COREMOD_JOB_BUILDER:
-                    builders++;
-                    break;
-                case COM_MINECOLONIES_COREMOD_JOB_DELIVERYMAN:
-                    deliverymen++;
-                    break;
-                case COM_MINECOLONIES_COREMOD_JOB_MINER:
-                    miners++;
-                    break;
-                case COM_MINECOLONIES_COREMOD_JOB_FISHERMAN:
-                    fishermen++;
-                    break;
-                case COM_MINECOLONIES_COREMOD_JOB_LUMBERJACK:
-                    lumberjacks++;
-                    break;
-                case COM_MINECOLONIES_COREMOD_JOB_FARMER:
-                    farmers++;
-                    break;
-                case COM_MINECOLONIES_COREMOD_JOB_GUARD:
-                    guards++;
-                    break;
-                case COM_MINECOLONIES_COREMOD_JOB_BAKER:
-                    bakers++;
-                    break;
-                case COM_MINECOLONIES_COREMOD_JOB_COOK:
-                    cooks++;
-                    break;
-                case "":
-                    break;
-                default:
-                    workers++;
-            }
+            final int length = citizen.getJob().split("\\.").length;
+            final String job = citizen.getJob().split("\\.")[length  - 1].toLowerCase();
+            jobCountMap.put(job, ( jobCountMap.get(job) == null ? 1 : jobCountMap.get(job) + 1 ) );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS864)